Build failure: bluesnarfer-0.1

## Steps To Reproduce

Steps to reproduce the behavior:

1. nix build --impure github:nixos/nixpkgs/nixos-unstable#bluesnarfer

## Build log

<!-- insert build log in code block in collapsable section -->

<details>

<summary>Build Log</summary>

```
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/6q4z6rjm03f7v1vf0lfx1l73k935z2qh-source
source root is source/bluesnarfer
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
no configure script, doing nothing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build flags: SHELL=/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/bash
gcc -Iinclude -W -g3 -lbluetooth src/bluesnarfer.c -o bluesnarfer
src/bluesnarfer.c: In function 'main':
src/bluesnarfer.c:59:14: error: implicit declaration of function 'getuid'; did you mean 'tcgetsid'? [-Wimplicit-function-declaration]
   59 |         if ( getuid() ) {
      |              ^~~~~~
      |              tcgetsid
src/bluesnarfer.c: In function 'rfcomm_read':
src/bluesnarfer.c:247:36: error: passing argument 2 of 'getline' from incompatible pointer type [-Wincompatible-pointer-types]
  247 |                 r = getline(&line, &line_size, fp);
      |                                    ^~~~~~~~~~
      |                                    |
      |                                    int *
In file included from src/bluesnarfer.c:20:
/nix/store/kj8hbqx4ds9qm9mq7hyikxyfwwg13kzj-glibc-2.40-36-dev/include/stdio.h:698:46: note: expected 'size_t * restrict' {aka 'long unsigned int * restrict'} but argument is of type 'int *'
  698 |                           size_t *__restrict __n,
      |                           ~~~~~~~~~~~~~~~~~~~^~~
src/bluesnarfer.c: In function 'switch_cmd':
src/bluesnarfer.c:369:31: error: implicit declaration of function 'info_cmd' [-Wimplicit-function-declaration]
  369 |                         ret = info_cmd(fd);
      |                               ^~~~~~~~
src/bluesnarfer.c: In function 'search_cmd':
src/bluesnarfer.c:548:39: warning: too many arguments for format [-Wformat-extra-args]
  548 |                 snprintf(buffer, 256, "AT+CPBS=\"ME\"\r\n", options.phonebook);
      |                                       ^~~~~~~~~~~~~~~~~~~~
make: *** [Makefile:11: bluesnarfer] Error 1
```

</details>

## Additional context

<!-- Add any other context about the problem here. -->

Last (successful) nixos rebuild was before Xmas.

## Metadata

<!-- Please insert the output of running `nix-shell -p nix-info --run "nix-info -m"` below this line -->
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.64, NixOS, 25.05 (Warbler), 25.05.20241217.d3c42f1`
 - multi-user?: `yes`
 - sandbox: `relaxed`
 - version: `nix-env (Nix) 2.24.11`
 - nixpkgs: `/nix/store/22r7q7s9552gn1vpjigkbhfgcvhsrz68-source`


## Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->
@fgaz 

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/bl/bluesnarfer/package.nix
+++ b/pkgs/by-name/bl/bluesnarfer/package.nix
@@ -21,6 +21,11 @@ stdenv.mkDerivation (finalAttrs: {
 
   strictDeps = true;
 
+  env.NIX_CFLAGS_COMPILE = toString [
+    "-Wno-incompatible-pointer-types"
+    "-Wno-implicit-function-declaration"
+  ];
+
   installPhase = ''
     runHook preInstall
     install -Dm755 bluesnarfer $out/bin/bluesnarfer

